### ISSUE TITLE
Scoping issue with const/let and await

### ISSUE THREAD

**Issue by assaf** [6to5/6to5#565]
Minimal test case:

```js
async function fail() {
 if (true) {
    let x = await Promise.resolve(1);
    function f() {
      return x;
    }
  }
}
```

The error I get is:

```
                var x = await Promise.resolve(1);
                              ^^^^^^^
SyntaxError: Unexpected identifier
```

This happens when:

- Variable declared inside a scope (e.g. condition, for/loop)
- Declared with local scope (`const` or `let`)
- Assigned the result of an expression using `await`
- Referenced from another function within that scope

**Comment by sebmck** [comment]
Similar issue to #564, async functions will need to be handled the same way as yield currently is throughout the codebase, ie. `await*` and creating async functions for closure wrappers.

**Comment by sebmck** [comment]
Fixed in the 3.0.0 branch.

**Comment by sebmck** [comment]
Fixed as of 3.0.0
